File: src/features/ai-platform/infra/commands/mcp.ts
import { invoke } from '@tauri-apps/api/core';

export interface McpSyncApps {
  claude: boolean;
  codex: boolean;
  gemini: boolean;
  opencode: boolean;
}

export interface McpServerRecord {
  id: string;
  name: string;
  command: string;
  args: string[];
  env: Record<string, string>;
  syncApps: McpSyncApps;
  enabled: boolean;
}

export interface McpSnapshot {
  servers: McpServerRecord[];
  templates: McpServerRecord[];
  source: string;
  validations: McpServerValidation[];
  syncStatuses: McpAppSyncStatus[];
}

export interface McpSyncResult {
  status: string;
  message: string;
  syncedApps: string[];
  appResults: McpAppSyncStatus[];
}

export interface McpServerValidation {
  serverId: string;
  status: string;
  issues: string[];
}

export interface McpAppSyncStatus {
  app: string;
  status: string;
  configPath: string;
  exists: boolean;
  targetedServers: number;
  syncedServers: number;
  message: string;
}

export function getAiPlatformMcpSnapshot() {
  return invoke<McpSnapshot>('ai_platform_get_mcp_snapshot');
}

export function saveAiPlatformMcpServer(server: McpServerRecord) {
  return invoke<McpSnapshot>('ai_platform_save_mcp_server', { server });
}

export function deleteAiPlatformMcpServer(serverId: string) {
  return invoke<McpSnapshot>('ai_platform_delete_mcp_server', { serverId });
}

export function syncAiPlatformMcpServers() {
  return invoke<McpSyncResult>('ai_platform_sync_mcp_servers');
}

File: src-tauri/src/ai_platform/interfaces/dto/mcp.rs
use crate::ai_platform::domain::mcp::McpServerRecord;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct McpServerValidationDto {
    pub server_id: String,
    pub status: String,
    pub issues: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct McpAppSyncStatusDto {
    pub app: String,
    pub status: String,
    pub config_path: String,
    pub exists: bool,
    pub targeted_servers: usize,
    pub synced_servers: usize,
    pub message: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct McpSnapshotDto {
    pub servers: Vec<McpServerRecord>,
    pub templates: Vec<McpServerRecord>,
    pub source: String,
    pub validations: Vec<McpServerValidationDto>,
    pub sync_statuses: Vec<McpAppSyncStatusDto>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct McpSyncResultDto {
    pub status: String,
    pub message: String,
    pub synced_apps: Vec<String>,
    pub app_results: Vec<McpAppSyncStatusDto>,
}